### PR TITLE
convert new signal

### PR DESCRIPTION
**Pull request by snopoke**
@millerdev

### CODE REVIEW

File: fluff/app_config.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.apps import AppConfig
+from django.db.models.signals import post_migrate
+from fluff.signals import catch_signal
+
+
+class FluffAppConfig(AppConfig):
+    name = 'fluff'
+
+    def ready(self):
+        post_migrate.connect(catch_signal, sender=self)
```
Comment (snopoke) [inline fluff/app_config.py]: Doing this means the signal only gets fired once (for this app) instead of for every app.

File: fluff/__init__.py [hunk 2]
```diff
@@ -15,6 +15,9 @@
 import fluff.util
 
 
+default_app_config = 'fluff.app_config.FluffAppConfig'
```
Comment (czue) [inline fluff/__init__.py]: neat (had to go read [the docs](https://docs.djangoproject.com/en/1.8/ref/applications/#configuring-applications) to figure out what this was doing)

Comment (millerdev) [inline fluff/__init__.py]: magic!

Comment (snopoke) [inline fluff/__init__.py]: yea, I didn't really want to go this route but also didn't want to have to coordinate with an update to settings.py

File: fluff/app_config.py (hunk 1 above)
Comment (esoergel) [inline fluff/app_config.py]: Curious if you think we need to worry about [this](https://docs.djangoproject.com/en/1.8/topics/signals/#connecting-receiver-functions):
> The ready() method may be executed more than once during testing, so you may want to [guard your signals from duplication](https://docs.djangoproject.com/en/1.8/topics/signals/#preventing-duplicate-signals), especially if you’re planning to send them within tests.

Comment (snopoke) [inline fluff/app_config.py]: I don't think so but could be good practice to do it anyway.

On 19 October 2015 at 19:46, Ethan Soergel <notifications@github.com> wrote:

> In fluff/app_config.py
> <https://github.com/dimagi/fluff/pull/105#discussion_r42402361>:
>
> > @@ -0,0 +1,10 @@
> > +from django.apps import AppConfig
> > +from django.db.models.signals import post_migrate
> > +from fluff.signals import catch_signal
> > +
> > +
> > +class FluffAppConfig(AppConfig):
> > +    name = 'fluff'
> > +
> > +    def ready(self):
> > +        post_migrate.connect(catch_signal, sender=self)
>
> Curious if you think we need to worry about this
> <https://docs.djangoproject.com/en/1.8/topics/signals/#connecting-receiver-functions>
> :
>
> The ready() method may be executed more than once during testing, so you
> may want to guard your signals from duplication
> <https://docs.djangoproject.com/en/1.8/topics/signals/#preventing-duplicate-signals>,
> especially if you’re planning to send them within tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/fluff/pull/105/files#r42402361>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Pull request merged**